src/Illuminate/Collections/Arr.php

```diff
@@ -708,15 +694,24 @@ public static function set(&$array, $key, $value)
      */
     public static function shuffle($array, $seed = null)
     {
-        if (is_null($seed)) {
-            shuffle($array);
-        } else {
+        if (! is_null($seed)) {
             mt_srand($seed);
             shuffle($array);
             mt_srand();
+
+            return $array;
         }
 
-        return $array;
+        $keys = array_keys($array);
+        $shuffled = [];
+
+        for ($i = count($keys) - 1; $i >= 0; $i--) {
```

Comment:
Given a zero-based array, the Fisher-Yates algorithm iterates from `count - 1` down to `1`.

Therefore, it should be: `for ($i = count($keys) - 1; $i >= 1; $i--) {`

Reply:
Ah, I just noticed that in https://github.com/laravel/framework/commit/9f39ac66aa125956eaf0a3c5fa67ee2209d51ae7 you tweaked the original algorithm, so I can't say for sure…

Reply:
Yep, the core counts down to `1` not `0`, but since it shuffles in-place while I'm injecting into a new array to preserve keys, the final element was always dropped off the end. This change ensures the final element is added into the new array.

Since it shouldn't be preserving keys, I will go back to the original algorithm.